### DISCUSSION TITLE
Switch from `jest.config.js` to bun?

### DISCUSSION THREAD

**Discussion by arnmishra** [oven-sh/bun#3849] [Q&A]
I've heard that bun test is blazing fast but haven't seen any information on how to port my setup over from jest. I have the following setup:

I have a `jest.config.js` that sets up all of my environment. It includes a `globalSetup: path.join(__dirname, 'jest.global-setup.js')`, and a `setupFilesAfterEnv: [path.join(__dirname, 'jest.setup.ts')]`,

`jest.global-setup.js` sets up my docker container with my postgres database set  to be a template to create per unit test dbs and then injects doppler secrets. 

`jest.setup.ts` creates per test postgres dbs from the template in a `beforeAll` and then creates a prisma client connection to the db

What I'd like to do is to run my `jest.global-setup.js` followed by my `jest.setup.ts` for all my UTs (all written in jest syntax). However, the only similar setup I see is the bun preload setup which doesn't offer a 1 for 1 option for what I want do. Any suggestions?

My full `jest.config.js`:
```
const config = {
  testEnvironment: 'node',
  moduleDirectories: ['node_modules', 'src', 'scripts'],
  testPathIgnorePatterns: ['/node_modules/', '/build/', '/.typecheck/'],
  globalSetup: path.join(__dirname, 'jest.global-setup.js'),
  testTimeout: 120_000,
  setupFilesAfterEnv: [path.join(__dirname, 'jest.setup.ts')],
  extensionsToTreatAsEsm: ['.ts'],
  transform: {
    '^.+\\.(t|j)sx?$': ['@swc/jest', swcConfig],
  },
  moduleNameMapper: {
    '^(\\.{1,2}/.*)\\.js$': '$1',
  },
}
```

**Comment by lpoulter1** [comment]
a jest migration guide would be great.
